File: python/tests/benchmarking.py
import time

from dotenv import load_dotenv

from e2b_code_interpreter.code_interpreter_sync import Sandbox

load_dotenv()

iterations = 10
create_sandbox_time = 0
first_exec_time = 0
second_exec_time = 0

for i in range(iterations):
    print("Iteration:", i + 1)
    start_time = time.time()
    sandbox = Sandbox()
    create_sandbox_time += time.time() - start_time

    start_time = time.time()
    sandbox.notebook.exec_cell("x = 1")
    first_exec_time += time.time() - start_time

    start_time = time.time()
    result = sandbox.notebook.exec_cell("x+=1; x")
    second_exec_time += time.time() - start_time

    sandbox.close()


print(f"Average Create Sandbox Time: {create_sandbox_time / iterations}s")
print(f"Average Execute Python x = 1 Time: {first_exec_time / iterations}s")
print(f"Average Execute Python x+=1; x Time: {second_exec_time / iterations}s")

File: js/tests/benchmarking.js
const { CodeInterpreter } = require('../dist')
const dotenv = require('dotenv')
dotenv.config()

const iterations = 10
let createSandboxTime = 0
let fistExecTime = 0
let secondExecTime = 0

async function main() {
  for (let i = 0; i < iterations; i++) {
    console.log('Iteration:', i + 1)
    let startTime = new Date()
    const sandbox = await CodeInterpreter.create()
    createSandboxTime += new Date() - startTime

    startTime = new Date()
    await sandbox.notebook.execCell('x = 1')
    fistExecTime += new Date() - startTime

    startTime = new Date()
    const result = await sandbox.notebook.execCell('x+=1; x')
    secondExecTime += new Date() - startTime

    await sandbox.close()
  }
  console.log('Average create sandbox time:', createSandboxTime / iterations)
  console.log('Average first exec time:', fistExecTime / iterations)
  console.log('Average second exec time:', secondExecTime / iterations)
}
main().catch(console.error)
